Cars-ut.cpp

```diff
@@ -0,0 +1,54 @@
+#include "Car.hpp"
+#include "ElectricCar.hpp"
+#include "ElectricEngine.hpp"
+#include "HybridCar.hpp"
+#include "PetrolCar.hpp"
+#include "exceptions.hpp"
+
+#include "gtest/gtest.h"
+
+class CarsTest : public ::testing::Test {
+protected:
+    CarsTest()
+        : electricCar_(std::make_shared<ElectricEngine>(130, 650)),
+          petrolCar_(std::make_shared<PetrolEngine>(120, 1600, 6)),
+          hybridCar_(std::make_shared<PetrolEngine>(150, 700, 5),
+                     std::make_shared<ElectricEngine>(150, 600)) {}
+
+    ElectricCar electricCar_;
+    PetrolCar petrolCar_;
+    HybridCar hybridCar_;
+};
+
+TEST_F(CarsTest, PetrolCarThrowRearGearExceptionWhenChangeFrom4ToMinus1) {
+    petrolCar_.getPetrolEngine()->changeGear(2);
+    ASSERT_THROW(petrolCar_.getPetrolEngine()->changeGear(-1), RearGearChangeException);
```

Comment:
you can assign an engine to a variable and refer via it
```suggestion
    auto engine = petrolCar_.getPetrolEngine();
    engine->changeGear(2);
    ASSERT_THROW(engine->changeGear(-1), RearGearChangeException);
```

Reply:
Done.